src/structure/group.rs

```diff
@@ -35,3 +38,45 @@ pub trait Group<O: Op>
 {}
 
 impl_marker!(Group<Additive>; i8, i16, i32, i64);
+
+/*
+ *
+ * A subgroup trait inherit from its parent groups.
+ *
+ */
+
+/// The group `E(n)` of isometries, i.e., rotations, reflexions, and translations.
+pub trait EuclideanGroupApprox<S: RealApprox, E: EuclideanSpaceApprox<S>>: GroupApprox<Multiplicative> {
+    /// Applies this group's action on a point from the euclidean space.
+    fn transform_point(&self, pt: &E) -> E;
+
+    /// Applies this group's action on a vector from the euclidean space.
+    ///
+    /// If `v` is a vector and `a, b` two point such that `v = a - b`, the action `∘` on a vector
+    /// is defined as `self ∘ v = (self × a) - (self × b)`.
+    fn transform_vector(&self, v: &E::Vector) -> E::Vector;
+}
+
+/// The group `SE(n)` of orientation-preserving isometries, i.e., rotations and translations.
+///
+/// This is a subgroup of `E(n)`.
+pub trait SpecialEuclideanGroupApprox<S: RealApprox, E: EuclideanSpaceApprox<S>>: EuclideanGroupApprox<S, E> {
+}
+
+/// The group `T(n)` of translations.
+///
+/// This is a subgroup of `SE(n)`.
+pub trait TranslationGroupApprox<S: RealApprox, E: EuclideanSpaceApprox<S>>: SpecialEuclideanGroupApprox<S, E> {
```

Comment:
I don't have any link but the proof is straightforward. Assuming you accept that SE(n) is itself a group, we only have to check that T(n) is stable under composition and inversion.

SE(n) represents rotations followed by translations. Elements of T(n) can be seen as elements of SE(n) with rotations equal to the identity. Those are called "pure translations".

* It is clear that the composition of two pure translations is still a pure translation.
* The inverse of a pure translation is also a pure translation (its negation).
* The identity element of T(n) (again, seeing elements of T(n) as elements of SE(n)) is the identity rotation followed by the zero translation. Thus the identity element of T(n) is the same as the one of SE(n).

That's all you need to prove that T(n) is a subgroup of SE(n).

I agree that the mathematical naming may be confusing. The `S` prefix added to `O` and `E` to form `SO` and `SE` means `Special` (that is, Special-Euclidean and Special-Orthogonal`). This signify that the linear part of the transformations has a determinant equal to `1` (and is positive).